spec/feature/posts_show_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show page', type: :feature do
+  let!(:user) { User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 10) }
+  let!(:post) do
+    Post.create(
+      title: 'Test Post',
+      text: 'This is a test post.',
+      author_id: user.id,
+      comments_counter: 3,
+      likes_counter: 7
+    )
+  end
+
+  describe 'GET #show' do
+    before { visit user_post_path(post) }
+
+    it 'displays the post title' do
+      expect(page).to have_content(post.title)
+    end
+
+    it 'displays the number of comments' do
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+
+    it 'displays the number of likes' do
+      expect(page).to have_content("Likes: #{post.likes_counter}")
+    end
+
+    it 'displays the post body' do
+      expect(page).to have_content(post.text)
+    end
+
+    it 'displays the username of each commenter and their comment' do
+      post.comments.each do |comment|
+        expect(page).to have_content("#{comment.user.name} - #{comment.text}")
+      end
+    end
+  end
+end
```

Comment:
- [ ] For the `Posts#Show` page, you forgot to write tests that confirm the following:

- **_Tests that assert you can see who wrote the post._**
- **_Tests that assert you can see the comment each commentor left._**

Kindly write these tests. 👍 😎 

![Screenshot_4](https://github.com/CaptainBawa/Blog-App/assets/77106962/5a22af62-9279-4bda-a46e-290437f4c225)

Reply:
I have included all the test cases as per your request. Additionally, I've incorporated a test case to verify the last suggestion. Please review the attached screenshot to confirm this.

![Capture](https://github.com/CaptainBawa/Blog-App/assets/82339317/226f31a5-f279-4304-8fcd-3038ff1e8a89)

It checks to verify the name of the person who wrote the comment with their comments attached.